First 3rd party library doc

kariharju opened the pull request.

kariharju committed:
Draft for first 3rd party library doc

kariharju committed:
Added download -snippet

cmin764 commented on 3rd_party/requests/README.md:
```diff
@@ -0,0 +1,29 @@
+# Requests -library
+
+Requests -library is one of the most known and used Python libraries.
```
```suggestion
# Requests

The `requests` library is one of the most known and used Python libraries for HTTP calls.
```

cmin764 commented on 3rd_party/requests/README.md:
```diff
@@ -0,0 +1,29 @@
+
+👉 Their slogan tells it all: `HTTP for Humans`
+
+Whether you need to download a file or check if a website is responding, for everything HTTP, you should just use `requests.`
```
```suggestion
Whether you need to download a file, interact with an API or simply checking if a website is responding, for everything HTTP, you should just use `requests`.
```

cmin764 commented on 3rd_party/requests/README.md:
````diff
@@ -0,0 +1,29 @@
+```
+
+Examples in Robocorp Portal:
+* [Python PLaywright -template](https://robocorp.com/portal/robot/robocorp/template-python-browser)
````
This example is almost right, as it needs to be refactored in order to use the `requests` library instead.

cmin764 replied:
Done with [PR](https://github.com/robocorp/template-python-browser/pull/3).

cmin764 commented on 3rd_party/requests/README.md:
````diff
@@ -0,0 +1,29 @@
+    return local_filename
+```
+
+Examples in Robocorp Portal:
````
Right before this "Examples" line, I would also add an entry called "Snippets", where we reference every snippet we built, so we can list them all along with a description. (and for also highlighting the `[snippets](./snippets)` folder itself)

cmin764 reviewed (approved):
This looks good, left a couple of potential improvements, which I will apply as soon as I refactor the referenced Portal example in order to use `requests` instead of `RPA.HTTP`.

cmin764 committed:
Reword some statements to make things clearer

cmin764 committed:
Merge branch 'master' into 3rd-party-docs

cmin764 commented on 3rd_party/requests/README.md:
```diff
@@ -0,0 +1,29 @@
+
+* [PyPI](https://pypi.org/project/requests/)
+* [Documentation](https://requests.readthedocs.io/en/latest/)
+* [API reference](https://requests.readthedocs.io/en/latest/api/)
```
Also is best to use `-` for bullet points, as they stand out better from the asterisk (`*`) that implies text emphasising or bolding.

cmin764 committed:
Applied review comments

- better READMEs
- fixed bullet points
- small code example refactoring
- addition of snippets reference

cmin764 committed:
Move 3rd_party libs docs under the main docs dir
